src/material/core/ripple/ripple-renderer.ts

```diff
@@ -438,6 +451,19 @@ export class RippleRenderer implements EventListenerObject {
       }
     }
   }
+
+  /** Fix coordinates emitted from the trigger element when is not the container element. */
+  private _fixCoordinatesEmittedFromTrigger(x: number, y: number): [number, number] {
+    if (this._triggerElement && this._triggerElement !== this._containerElement) {
+      const containerRect = (this._containerRect =
+        this._containerRect || this._containerElement.getBoundingClientRect());
+      const triggerRect = (this._triggerRect =
+        this._triggerRect || this._triggerElement.getBoundingClientRect());
+      x = ((x - triggerRect.left) / triggerRect.width) * containerRect.width + containerRect.left;
+      y = ((y - triggerRect.top) / triggerRect.height) * containerRect.height + containerRect.top;
```

Comment:
This assumes that the dimensions of the trigger are translatable to the container right? I wonder if we would rather want to prevent the case where the trigger is outside of the container? What do you think?

Reply:
> This assumes that the dimensions of the trigger are translatable to the container right?

That's right. And I don't see any case where the dimensions of the trigger aren't translatable to the container because here we do simple proportion calculations (except when the [container height or width equal to 0 but this case is already covered](https://github.com/angular/components/blob/18.1.0-next.1/src/material/core/ripple/ripple-renderer.ts#L172-L173)).

> I wonder if we would rather want to prevent the case where the trigger is outside of the container?

No, I don't want to prevent the case where the trigger is outside of the container. I want to translate a position clicked in trigger element into the ripple container.

In this [stackblitz to reproduce the current behavior](https://stackblitz.com/edit/components-issue-gtvxae), I apply `[matRippleRadius]="10"` on the ripple container. But if you don't apply a radius, you can see that the ripple element coords is outside the ripple container (I clicked where there is the letter "X" in red) and create a huge radius:

![image](https://github.com/angular/components/assets/977713/9f58c4c7-6ccf-49b8-b029-947ca5c742b0)

In this example, the experience is not too shocking because the trigger is near to the ripple container. But in case where the trigger is far from the ripple container: the ripple element rendering is strange (when we think about user experience). It's a better user experience when the ripple element coords is inside the ripple container in coords translated from the original coords (where the user clicked on the trigger).

Note: indeed, this translation has no effect when the ripple element is centered.

Reply:
> That's right. And I don't see any case where the dimensions of the trigger aren't translatable to the container because here we do simple proportion calculation

Right, but does it make sense to do a simple proportion calculation? what's the use-case? I.e. in most cases the ripple should originate where the click happens, and different proportions wouldn't necessarily make sense then, right?

> No, I don't want to prevent the case where the trigger is outside of the container. I want to translate a position clicked in trigger element into the ripple container.

What's the use-case though? Why wouldn't the trigger not be inside the container? or a sibling that is "positioned over the ripple container"?

> In this [stackblitz to reproduce the current behavior](https://stackblitz.com/edit/components-issue-gtvxae), I apply [matRippleRadius]="10" on the ripple container. But if you don't apply a radius, you can see that the ripple element coords is outside the ripple container (I clicked where there is the letter "X" in red) and create a huge radius:

Yeah, this is a bug. Good catch, and one solution would be to disallow the trigger to be outside of the boundaries of the container I think.

Reply:
> Yeah, this is a bug. Good catch, and one solution would be to disallow the trigger to be outside of the boundaries of the container I think.

I don't think that is a bug. As we can read in [ripple.md](https://github.com/angular/components/blob/main/src/material/core/ripple/ripple.md#ripple-trigger): it's a feature "to show ripples on interaction with some other element" that the container.

> Right, but does it make sense to do a simple proportion calculation? what's the use-case?

Check yourself: I made [another stackblitz](https://stackblitz.com/edit/components-issue-ratgfs?file=src%2Fexample-component.ts) and compare the user experience with the both triggers (I think that a demonstration can be more explicit than words).

Reply:
> I don't think that is a bug. As we can read in [ripple.md](https://github.com/angular/components/blob/main/src/material/core/ripple/ripple.md#ripple-trigger): it's a feature "to show ripples on interaction with some other element" that the container.

yeah, I actually created the implementation of ripple a couple of years ago. The intent was to support triggers of different elements, but not necessarily disconnected from the container visually (but to support the trigger overlaying the ripple elements, to e.g. not prevent further clicks).

We can limit this feature, or clarify the intent to address this problem. It's just an option we can consider. We can certainly also fix this if it makes sense to do so.

> Check yourself: I made [another stackblitz](https://stackblitz.com/edit/components-issue-ratgfs?file=src%2Fexample-component.ts) and compare the user experience with the both triggers (I think that a demonstration can be more explicit than words).

I follow what you are trying to achieve, but what is the real-world use-case here? Why would you want to display the ripple in a totally different element? Ripples from the Material specification generally don't seem to mention such a scenario: https://m2.material.io/develop/ios/supporting/ripple

Also cc. @crisbeto / @andrewseguin

Reply:
> I follow what you are trying to achieve, but what is the real-world use-case here? Why would you want to display the ripple in a totally different element?

It's subjective. The first time I use a trigger element different than the ripple container element (where the trigger element is far from the ripple container element): I don't understand why the ripple effect seems to be strange (the ripple effect was too different than the case when the ripple container element is also the trigger element). I needed to investigate to understand why the difference.

After that, I wrote a code to adapt the ripple position like in this [stackblitz](https://stackblitz.com/edit/components-issue-gtvxae). And later I made this pull request.

I would have been curious to have other opinions than ours. But if you want we can close this topic.